common/djangoapps/nodebb/signals/handlers.py

```diff
@@ -0,0 +1,42 @@
+from logging import getLogger
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from common.lib.nodebb_client.client import NodeBBClient
+from xmodule.modulestore.django import modulestore
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from student.models import CourseEnrollment
+
+log = getLogger(__name__)
+
+
+@receiver(post_save, sender=CourseOverview, dispatch_uid="nodebb.signals.handlers.create_category_on_nodebb")
+def create_category_on_nodebb(sender, instance, created, **kwargs):
+    if created:
+        status_code, response_body = NodeBBClient().categories.create(name=instance.display_name)
+
+        if status_code != 200:
+            log.error(
+                "Error: Can't create category for the given course {} due to {}".format(instance.id, response_body)
+            )
+
+        log.info('Success: Community created for course {}'.format(instance.id))
+
+
+@receiver(post_save, sender=CourseEnrollment, dispatch_uid="nodebb.signals.handlers.join_group_on_nodebb")
+def join_group_on_nodebb(sender, instance, created, **kwargs):
+    if created:
```

Comment:
I think this handler should call a celery task and that task should make relevant API calls to nodebb